### PR TITLE
Add TypeScript types

### LINKED ISSUE #5: TypeScript Declarations
Interested in PR with `index.d.ts`?

I'm currently using this:

```ts
declare module 'is-port-reachable' {
  export interface IsPortReachableOptions {
    timeout?: number | undefined;
    host?: string;
  }
  export default function(
    port: number | undefined,
    options?: IsPortReachableOptions
  ): Promise<boolean>;
}

```

### PR DESCRIPTION
**Pull request by mikicho**
Add typescript definition file to solve ts(7016)
Fixes #5

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks for creating a TypeScript definition. 🙌 Can you follow [this styleguide](https://github.com/sindresorhus/typescript-definition-style-guide)?

Some examples:
- https://github.com/sindresorhus/semver-regex/commit/039944bb5b3876913171b895488874b5c200e458
- https://github.com/sindresorhus/global-dirs/commit/0aae4d64887f77af94c4fb7ca12aa9f93c0a8291
- https://github.com/sindresorhus/md5-hex/commit/bbbfeff840f26973fbffc16d4863ae6d120079f3

**Comment by mikicho** [comment]
@sindresorhus done.
Please let me know if I missed something.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+Check if a local or remote port is reachable
+@example
+```
+import isPortReachable from 'is-port-reachable';
+
+if(await isPortReachable(3000)) {
+	// start server
+}
```
Comment (sindresorhus) [inline index.d.ts:26]: Use the same example as the readme.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+```
+*/
+declare function isPortReachable(port: number, opts?: {timeout?: number, host?: string}): Promise<boolean>;
```
Comment (sindresorhus) [inline index.d.ts]: Options should follow the style here: https://github.com/sindresorhus/typescript-definition-style-guide#documentation

**Comment by mikicho** [comment]
@sindresorhus  done

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+interface IsPortReachableOptions {
+  /**
+  Milliseconds to wait before giving up.
+
+  @default 1000
+  */
+  timeout?: number;
```
Comment (sindresorhus) [inline index.d.ts]: Use tab-indentation.

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  /**
+  Can be a domain or an IP.
+
+  @default localhost
```
Comment (sindresorhus) [inline index.d.ts]:
```suggestion
  @default 'localhost'
```

File: package.json [hunk 5]
```diff
@@ -16,10 +16,11 @@
 		"node": ">=8"
 	},
 	"scripts": {
-		"test": "xo && ava"
+		"test": "xo && ava && tsd"
 	},
 	"files": [
-		"index.js"
+		"index.js",
+    "index.d.ts"
```
Comment (sindresorhus) [inline package.json]: Incorrect indentation

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+interface IsPortReachableOptions {
```
Comment (sindresorhus) [inline index.d.ts]:
```suggestion
interface Options {
```

**Comment by sindresorhus** [comment]
The pull request needs a proper title.

**Comment by mikicho** [comment]
@sindresorhus done

**Pull request merged**